File: singleton/singleton.py
class Singleton:
    _instance = None

    @property
    def nome(self):
        return self._nome

    @nome.setter
    def nome(self, valor):
        self._nome = valor

    @staticmethod
    def get_instance():
        if not Singleton._instance:
            Singleton._instance = Singleton()
        return Singleton._instance


if __name__ == '__main__':
    obj1 = Singleton.get_instance()
    obj1.nome = 'Luiz'
    print(f'nome obj1: {obj1.nome}')

    obj2 = Singleton.get_instance()
    print(f'nome obj2: {obj2.nome}')

File: singleton/singleton.java
class Singleton {

    private static Singleton single_instance = null;
    public String s;

    private Singleton() {
        s = "Luiz Filipy";
    }

    public static Singleton getInstance() {
        if (single_instance == null)
            single_instance = new Singleton();
        return single_instance;
    }
}

class Main {
    public static void main(String args[]) {
        Singleton x = Singleton.getInstance();
        System.out.println("s " + x.s);

        Singleton y = Singleton.getInstance();
        System.out.println("s " + y.s);

        x.s = "Filipy";

        System.out.println("s " + x.s);
        System.out.println("s " + y.s);
    }
}